### DISCUSSION TITLE
Does RxTelegram have FSM?

### DISCUSSION THREAD

**Discussion by HedgehogNSK** [RxTelegram/RxTelegram.Bot#38] [Q&A] [answered]
Does RxTelegram have inside Finitie State Machine or any sort of classes that can help to manage bot states?

**Comment by niklasweimann** [comment, marked as answer]
Hey @HedgehogNSK 👋
Thanks for your Question. Unfortunately RxTelegram does not provide such functionality. Our main focus is to provide an API-Wrapper for the Telegram bot Api. What features would you expect from a bot state management?

**Reply by HedgehogNSK** [reply]
Probably I was to naive when I expected this functionality. I'm wirting my first tg bot. There are a lot of tutorials and examples in Python in an internet, And unfortunately  C# lacks deep and detailed examples. Therefore, I have to learn Telegram API by python examples. And I found out that in iogram has implementation of FSM inside the library. That's why I was expecting something like that in RxTelegram.  
The main benefit from those FSM is that you can create commands with complex behaviour that includes several interaction steps with the same user.

**Reply by niklasweimann** [reply]
We have a repository with examples, where you can find some example bots: [RxTelegram.Bot.Examples](https://github.com/RxTelegram/RxTelegram.Bot.Examples )

This feature sounds interesting to me and fun to build. I will have a look and maybe publish such a system as an extension for RxTelegram.Bot, but I definitely do not want to include it in the main project.

**Reply by HedgehogNSK** [reply]
Yes. I looked that repository before start. It would be nice to see there a complex examples, at least one, that demonstrates case of long interaction chain with a user. Usually in other libs tutors use as an example a short user data questionnaire. If it's possible to create react chain for such case, it would be awesome.

**Reply by niklasweimann** [reply]
Sure that's definitely possible. I will create such an example and get back to you as soon as it is available in the examples repository. ☺️✌️

**Reply by niklasweimann** [reply]
Hey @HedgehogNSK  👋 
The Example is now available in the Repository: [RxTelegram.Bot.Examples.Converstion](https://github.com/RxTelegram/RxTelegram.Bot.Examples/tree/master/Conversation).
 For more information about the finite state machine used in the example please have a look at the ReadMe file [here](https://github.com/RxTelegram/RxTelegram.Bot.Examples/tree/master/Conversation/StateMachine).
 
 If you need any more help please feel free to contact us again 😊✌️

**Reply by HedgehogNSK** [reply]
Sweeet.. 🔥 Thank you! ^_^
